controllers/watcher_controller.go

```diff
@@ -718,14 +725,69 @@ func (r *WatcherReconciler) createSubLevelSecret(
 	Log.Info(fmt.Sprintf("Creating SubCr Level Secret for '%s'", instance.Name))
 	databaseAccount := db.GetAccount()
 	databaseSecret := db.GetSecret()
+
+	secretName := instance.Name
+
+	//
+	// Get correct prometheus host, port and if TLS should be used
+	//
+	var PrometheusHost string
+	var PrometheusPort int32
+	var PrometheusTLS bool
+	var PrometheusCaCert string
+
+	if instance.Spec.PrometheusHost == "" {
+		PrometheusHost = fmt.Sprintf("%s-prometheus.%s.svc", telemetryv1.DefaultServiceName, instance.Namespace)
+	} else {
+		PrometheusHost = instance.Spec.PrometheusHost
+	}
+	if instance.Spec.PrometheusPort == 0 {
+		PrometheusPort = telemetryv1.DefaultPrometheusPort
+	} else {
+		PrometheusPort = instance.Spec.PrometheusPort
+	}
+	if instance.Spec.PrometheusHost == "" {
+		// We're using MetricStorage for Prometheus. Set TLS accordingly
+		metricStorage := &telemetryv1.MetricStorage{}
+		err := r.Client.Get(ctx, client.ObjectKey{
+			Namespace: instance.Namespace,
+			Name:      telemetryv1.DefaultServiceName,
+		}, metricStorage)
+		if err != nil {
+			Log.Info(fmt.Sprintf("Metrics storage not found in '%s'", instance.Namespace))
+			instance.Status.Conditions.Set(condition.FalseCondition(
+				condition.InputReadyCondition,
+				condition.ErrorReason,
+				condition.SeverityError,
+				"Metrics storage not found in the controlplane",
+			))
+			return secretName, err
+		}
+
+		PrometheusTLS = metricStorage.Spec.PrometheusTLS.Enabled()
+	} else {
+		// We're using user-deployed Prometheus. Set TLS based on PrometheusTLSCaCertSecret
+		PrometheusTLS = instance.Spec.PrometheusTLSCaCertSecret != nil
+	}
+
+	if instance.Spec.PrometheusTLSCaCertSecret != nil {
+		PrometheusCaCert = watcher.CustomPrometheusCaCertFolderPath + instance.Spec.PrometheusTLSCaCertSecret.Key
```

Comment:
Can we use [filepath.Join ](https://pkg.go.dev/path/filepath)to add these two file path. 
```suggestion
		PrometheusCaCert = filepath.Join(watcher.CustomPrometheusCaCertFolderPath , instance.Spec.PrometheusTLSCaCertSecret.Key)
```